content/blog/its-time-to-try-out-pixi/contents.lr

```diff
@@ -0,0 +1,995 @@
+within a single, canonical configuration file (e.g. `pyproject.toml`).
+
+### Software developer
+
+#### Publishing to PyPI
+
+As a Python tool developer,
+I create stuff that needs to be distributed to other Pythonistas.
+As such, I need to be able to leverage
+existing Python publishing tooling (e.g., PyPI or conda-forge).
+
+#### Running tests
+
+Whatever tool I use, I also need to be able to run software tests easily.
+Ideally, this would be done with one command, e.g., `pytest`, with minimal overhead.
+The software testing environment should be the same as my development environment.
+
+### Both personas' needs
+
+#### Simplified environment specification
+
+As mentioned above, I currently use `environment.yml` and `pyproject.toml`
+to specify runtime and development dependencies.
+Runtime dependencies are declared in `pyproject.toml`,
+while development dependencies are declared in `environment.yml`.
+However, this way of separating concerns means that
+we have duplication in our dependencies specification!
+Ideally, we'd like to do this with a single file.
+
+#### Tooling for fast environment builds
+
+Whether we are in a Docker container or not,
+I'd like to see much faster container and environment build times
+with tooling for caching when compared to what I currently experience.
+
+(Note to self:
+Product-oriented folks keep talking about "don't solution, tell me the problem",
+but I have to say -- I only really knew how much of a problem this was
+once I touched the solution I'm about to talk about!)
+
+#### Automatic updates of lock files
+
+When I add a dependency to my environment,
+I'd like to see the lock file automatically updated
+so that I don't have to remember to do that in the future.
+
+## How pixi operates
+
+With the desiderata outlined above as contextual information,
+let's now see how `pixi` works.
+
+### Installation
+
+What's awesome is that `pixi` is installed via a simple, one-bash-script install.
+Official instructions are available [here](https://pixi.sh/latest/#installation),
+but as of the time of writing, the only thing we need to run is:
+
+```bash
+curl -fsSL https://pixi.sh/install.sh | bash
+```
+
+This sets up `pixi` within your home directory
+and adds the `pixi` path to your [PATH environment variable](https://ericmjl.github.io/essays-on-data-science/software-skills/environment-variables/).
+
+I also recommend getting set up with autocompletion for your shell.
+Once again, official instructions are [here](https://pixi.sh/latest/#autocompletion),
+and what I used for the `zsh` (macOS' default shell) was:
+
+```bash
+echo 'eval "$(pixi completion --shell zsh)"' >> ~/.zshrc
+```
+
+### Starting a new project with `pixi` enabled
+
+For those who are used to `conda` idioms,
+`pixi` won't create a centralized `conda` environment
+but instead will create a `.pixi` sub-directory
+within the directory of your `pixi` configuration file
+(either `pyproject.toml` or `pixi.toml`).
```

Comment:
You could possibly add that this is working like:
- `npm`:  Nodejs ecosystem
- `cargo`: Rust ecosystem
- `poetry`: another pypi tool
If that would help your readers understand the paradigm

Reply:
That's a great point. I've just updated the text to allude to this point!